### PR TITLE
feat: domain model, database schema, and Design CRUD API

### LINKED ISSUE #1: Domain model and database schema
## Summary

Define the core Go domain types and initial SQLite database schema that underpin the entire application. This is the foundation every other feature builds on. Includes a minimal Design CRUD API to verify the full stack (Go → SQLite → Angular) end-to-end.

## Features / Scope

| Feature | Description |
|---------|-------------|
| Domain types | Go structs in `server/internal/models/` for all core entities |
| TypeScript mirrors | Matching interfaces in `frontend/src/app/models/` |
| SQLite setup | Database initialization with WAL mode for multi-tab safety |
| Migration tooling | golang-migrate with auto-rollback on failure |
| Initial schema | First migration creating all tables with FKs and indexes |
| Store layer | Repository pattern in `server/internal/store/` |
| Design CRUD API | Minimal `/api/designs` endpoints (create, list, get, delete) to validate full stack |

## Architecture / Design Notes

- **Module location**: Domain types in `server/internal/models/`, store layer in `server/internal/store/`, API handlers in `server/internal/api/handlers/`
- **Storage**: SQLite with WAL mode enabled for concurrent read access (multi-tab). Single-writer with automatic retry on `SQLITE_BUSY`.
- **Migration tooling**: golang-migrate — migrations are sequential numbered `.sql` files in `server/internal/migrations/` (e.g., `000001_initial_schema.up.sql`, `000001_initial_schema.down.sql`). Migrations auto-rollback on failure.
- **Key technical decisions**:
  - Use `modernc.org/sqlite` (pure Go, no CGO) for easier cross-compilation and npm distribution
  - Domain types use value objects where appropriate (e.g., `PortSpeed`, `FabricType` as typed enums)
  - Schema supports hierarchical aggregation: Device → Rack → Block → Super-Block → Site
  - Foreign keys enforced (`PRAGMA foreign_keys = ON`)
- **Security**: Not a primary concern (local-first), but validate all user input at API boundary

## Integration Points

- **Consumed by**: Every feature issue (#2–#10) depends on these domain types
- **Design CRUD API**: Provides the first working endpoint that #12 (app shell) can connect to
- **Store layer patterns**: Establishes the repository pattern that #2 (catalog), #3 (racks), #4 (fabrics) extend
- **Migration tooling**: All future schema changes (#2, #3, #4, #5, #6, #7) add migrations using this tooling
- **TypeScript interfaces**: Frontend features (#8, #9, #11, #12) import from `frontend/src/app/models/`

## Acceptance Criteria

- [ ] Core domain types defined in `server/internal/models/`: Design, Fabric (front-end and back-end), Rack (physical and logical), Device, Port, DeviceModel, Site, Block, SuperBlock
- [ ] TypeScript interfaces mirroring Go types in `frontend/src/app/models/`
- [ ] SQLite database initialization with WAL mode enabled
- [ ] golang-migrate integrated with auto-rollback on migration failure
- [ ] Initial migration creating all tables with foreign keys and indexes
- [ ] Database connection management in `server/internal/store/` with connection pooling for concurrent reads
- [ ] Minimal Design CRUD API: `POST /api/designs`, `GET /api/designs`, `GET /api/designs/:id`, `DELETE /api/designs/:id`
- [ ] Migration failure auto-rolls back — no partial schema state
- [ ] Concurrent read access works (multi-tab scenario: two browser tabs hitting the API simultaneously)
- [ ] Unit tests for migration up/down using in-memory SQLite
- [ ] Unit tests for migration rollback on simulated failure
- [ ] Unit tests for Design CRUD operations (create, get, list, delete, not-found)
- [ ] Unit tests for concurrent read access
- [ ] Tests pass, build succeeds

## Edge Cases

- Migration fails mid-way → auto-rollback, return clear error, DB remains in pre-migration state
- Concurrent writes from multiple tabs → SQLite busy handling with automatic retry (configurable timeout)
- `DELETE /api/designs/:id` on non-existent design → 404 with structured error response
- `GET /api/designs/:id` with invalid ID format → 400 with validation error
- Database file doesn't exist on first run → auto-create with all migrations applied
- Database file is locked by another process → clear error message, not a hang

## Documentation

- [ ] Data model diagram (ERD) in `docs/internal/specs/data-model.md`
- [ ] Migration conventions documented in `CLAUDE.md` (naming, up/down pattern, rollback behavior)
- [ ] API response format conventions documented in `CLAUDE.md`

## Dependencies

None — this is the foundational issue.

### PR DESCRIPTION
**Pull request by rnwolfe**
## Summary

- Defines core Go domain types in `server/internal/models/`: Design, Fabric (frontend/backend tiers), Site, SuperBlock, Block, Rack (physical/logical), Device, Port, DeviceModel
- Mirrors all types as TypeScript interfaces in `frontend/src/app/models/`
- Adds SQLite database with WAL mode and `PRAGMA foreign_keys=ON` using `modernc.org/sqlite` (pure Go, no CGO)
- Implements golang-migrate with embedded SQL migration files (`0001_initial_schema` up/down)
- Adds store layer (`DesignStore`) with repository pattern
- Adds service layer (`DesignService`) with domain error types (`ErrNotFound`, `ErrConstraintViolation`)
- Implements Design CRUD REST API: `POST /api/designs`, `GET /api/designs`, `GET /api/designs/{id}`, `DELETE /api/designs/{id}`
- Wires full stack in `main.go`: store → service → handler
- Adds data model ERD in `docs/internal/specs/data-model.md`

## Acceptance Criteria

- [x] Domain types defined: Design, Fabric (frontend+backend), Rack (physical+logical), Device, Port, DeviceModel, Site, Block, SuperBlock — all in `server/internal/models/`
- [x] TypeScript mirrors in `frontend/src/app/models/` for all types
- [x] SQLite with WAL mode (`PRAGMA journal_mode=WAL`) using `modernc.org/sqlite` (pure Go, no CGO)
- [x] golang-migrate with sequential numbered SQL files in `server/internal/migrations/`
- [x] Initial migration (`0001_initial_schema`) creates all tables with foreign keys and indexes
- [x] Down migration reverses the schema (drops all tables)
- [x] Store layer with repository pattern in `server/internal/store/`
- [x] Design CRUD API: `POST /api/designs`, `GET /api/designs`, `GET /api/designs/{id}`, `DELETE /api/designs/{id}`
- [x] Unit tests: migration up/down, rollback, CRUD operations, concurrent reads
- [x] Data model ERD in `docs/internal/specs/data-model.md`

## Test plan

- [x] `make test-server` — all Go tests pass (31 tests across store, service, handler packages)
- [x] `make build-server` — server binary builds successfully
- [x] `make lint-server` — `go vet` passes with no issues
- [x] `make test-frontend` — Angular unit tests pass (2 tests)
- [x] `make build-frontend` — Angular build succeeds

Closes #1

🤖 Generated with Claude Code

### CODE REVIEW

File: server/internal/store/db.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	if _, err := db.Exec("PRAGMA journal_mode=WAL"); err != nil {
+		db.Close()
+		return nil, fmt.Errorf("enable WAL: %w", err)
+	}
+
+	// Enforce referential integrity.
+	if _, err := db.Exec("PRAGMA foreign_keys=ON"); err != nil {
+		db.Close()
+		return nil, fmt.Errorf("enable foreign keys: %w", err)
+	}
```
[the remark this answers is not in the document]
Comment (rnwolfe) [reply server/internal/store/db.go]: Fixed in PR #17. Replaced the connection-scoped `PRAGMA foreign_keys=ON` exec with a DSN pragma (`?_pragma=foreign_keys(1)`), appending with `&` when the path already contains query parameters. Every connection the pool opens now has FK enforcement enabled from the start.

File: server/internal/service/design_service.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// NewDesignService returns a new DesignService backed by repo.
+func NewDesignService(repo DesignRepository) *DesignService {
+	return &DesignService{repo: repo}
+}
+
+// CreateDesign validates and creates a new Design.
+func (s *DesignService) CreateDesign(name, description string) (*models.Design, error) {
+	if name == "" {
+		return nil, fmt.Errorf("%w: design name is required", models.ErrConstraintViolation)
+	}
```
[the remark this answers is not in the document]
Comment (rnwolfe) [reply server/internal/service/design_service.go]: Fixed in PR #17. Added `name = strings.TrimSpace(name)` before the empty-string check, so whitespace-only names are rejected and the trimmed value is stored.
